skimage/measure/_regionprops.py

```diff
@@ -1089,6 +1101,166 @@ def regionprops(label_image, intensity_image=None, cache=True,
+           2010. Ecole Nationale Superieure des Mines de Saint-Etienne.
+           https://tel.archives-ouvertes.fr/tel-00560838
+    .. [2] Ohser J., Nagel W., Schladitz K. (2002) The Euler Number of
+           Discretized Sets - On the Choice of Adjacency in Homogeneous
+           Lattices. In: Mecke K., Stoyan D. (eds) Morphology of Condensed
+           Matter. Lecture Notes in Physics, vol 600. Springer, Berlin,
+           Heidelberg.
+
+    Examples
+    --------
+    >>> import numpy as np
+    >>> SAMPLE = np.zeros((100,100,100));
+    >>> SAMPLE[40:60, 40:60, 40:60]=1
+    >>> euler_number(SAMPLE) # doctest: +ELLIPSIS
+    1...
+    >>> SAMPLE[45:55,45:55,45:55] = 0;
+    >>> euler_number(SAMPLE) # doctest: +ELLIPSIS
+    2...
+    >>> SAMPLE = np.array([[0, 0, 0, 0, 0, 0, 0, 0, 0, 1, 0, 1, 1, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 0, 0, 0, 0, 0, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+    ...                    [0, 0, 0, 0, 0, 0, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0],
+    ...                    [1, 0, 1, 0, 0, 1, 1, 0, 1, 1, 0, 0, 1, 1, 1, 1, 1, 0],
+    ...                    [0, 1, 1, 1, 1, 1, 1, 1, 0, 1, 1, 0, 0, 0, 1, 1, 1, 1],
+    ...                    [0, 1, 1, 0, 0, 1, 1, 1, 0, 0, 0, 0, 0, 0, 0, 0, 1, 1]])
+    >>> euler_number(SAMPLE)  # doctest:
+    0
+    >>> euler_number(SAMPLE, connectivity=1)  # doctest:
+    2
+    """
+
+    # as image can be a label image, transform it to binary
+    image = (image > 0).astype(np.int)
+    image = np.pad(image, ((1, 1),), mode='constant')
+
+    # check connectivity
+    if connectivity is None:
+        connectivity = image.ndim
+
+    # config variable is an adjacency configuration. A coefficient given by
+    # variable coefs is attributed to each configuration in order to get
+    # the Euler characteristic.
+    if image.ndim == 2:
+
+        config = np.array([[0, 0, 0], [0, 1, 4], [0, 2, 8]])
+        if connectivity == 1:
+            coefs = [0, 1, 0, 0, 0, 0, 0,
+                     -1, 0, 1, 0, 0, 0, 0, 0, 0]
+        else:
+            coefs = [0, 0, 0, 0, 0, 0, -1,
+                     0, 1, 0, 0, 0, 0, 0, -1, 0]
+        bins = 16
+    else:  # 3D images
+        if connectivity == 2:
+            raise NotImplementedError(
+                    'For 3D images, Euler number is implemented '
+                    'for connectivities 1 and 3 only')
+
+        config = np.array([[[0, 0, 0], [0, 0, 0], [0, 0, 0]],
+                           [[0, 0, 0], [0, 1, 4], [0, 2, 8]],
+                           [[0, 0, 0], [0, 16, 64], [0, 32, 128]]])
+        coefs26 = np.array([0, 1, 1, 0, 1, 0, -2, -1,
```

Comment:
It would be great to pull these out of the function definition and name them `EULER_COEFS3D_26`. Next to the definition, a precise reference to how the values are generated would be handy. It's pretty cryptic! :joy: